client/src/components/History/providers/HistoryContentProvider/contentPayload.js

```diff
@@ -149,18 +152,13 @@ export const contentPayload = (cfg = {}) => {
             map(([[lastResponse, response], pos, hid]) => {
                 const updatesAtTop = response.maxHid > lastResponse.maxHid;
 
-                const scrollerInDefault = (pos.cursor == 0 && pos.key == null);
+                const scrollerExactlyAtTop = pos.cursor === 0 || pos.key === lastResponse.maxHid;
                 const fudge = 2;
-                const scrollNearLastTop = Math.abs(hid - lastResponse.maxContentHid) < fudge;
-                const scrollerAtTop = scrollerInDefault || scrollNearLastTop;
-
-                // console.group("should i reposition?");
-                // console.log("updatesAtTop", updatesAtTop);
-                // console.log("scrollerAtTop", scrollerAtTop);
-                // console.groupEnd();
+                const scrollNearLastTop = pos.key !== null ? Math.abs(pos.key - lastResponse.maxContentHid) < fudge : false;
+                const scrollerAtTop = scrollerExactlyAtTop || scrollNearLastTop;
 
                 if (updatesAtTop && scrollerAtTop) {
-                    return new ScrollPos({ cursor: 0, key: response.maxHid })
+                    return ScrollPos.create({ cursor: 0, key: response.maxContentHid });
                 }
                 return null;
             }),
```

Comment:
This is the code responsible for emitting the event that shifts the scroller back to the top after a content update.